pallets/api/src/messaging/mod.rs

```diff
@@ -250,6 +274,29 @@ pub mod pallet {
 		Messaging,
 	}
 
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberOf<T>> for Pallet<T> {
+		fn on_initialize(n: BlockNumberOf<T>) -> Weight {
```

Comment:
do we need to use a hook for this?

Wouldn't it be a little simpler to handle this lazily in `fn xcm_response`.

Basically, something along the lines of:
- store the timeout in `XcmQueries`, 
- in `xcm_response` compare the timeout from `XcmQueries` with the current block and then handle the message.

Reply:
One must also update and check the timeout if the message state is ever read otherwise it will be inconsistent.

This is impossible if a dapp is just reading storage directly and not going through the read API. 
Hence i think until https://github.com/paritytech/polkadot-sdk/pull/7896 is merged this will do as a temp replacement.